npm/iptm/iptm_test.go

```diff
@@ -223,123 +280,29 @@ func TestRun(t *testing.T) {
-	}
-
-	entry = &IptEntry{
-		Chain: util.IptablesForwardChain,
-		Specs: []string{
-			util.IptablesJumpFlag,
-			util.IptablesAzureChain,
-		},
-	}
-
-	// Ignore not exists errors
-	npmExists, _ = iptMgr.Exists(entry)
-
-	lineNum, err = iptMgr.GetChainLineNumber(util.IptablesAzureChain, util.IptablesForwardChain)
-	if err != nil {
-		t.Errorf("TestGetChainLineNumber @ initial iptMgr.GetChainLineNumber error: %s", err.Error())
-	}
-
-	switch {
-	case (npmExists && kubeExists):
-		if lineNum != 3 {
-			t.Errorf("TestGetChainLineNumber @ initial line number check iptMgr.GetChainLineNumber with npmExists: %t kubeExists: %t", npmExists, kubeExists)
-		}
-	case npmExists:
-		if lineNum == 0 {
-			t.Errorf("TestGetChainLineNumber @ initial line number check iptMgr.GetChainLineNumber with npmExists: %t kubeExists: %t", npmExists, kubeExists)
-		}
-	default:
-		if lineNum != 0 {
-			t.Errorf("TestGetChainLineNumber @ initial line number check iptMgr.GetChainLineNumber with npmExists: %t kubeExists: %t", npmExists, kubeExists)
-		}
-	}
+	fexec := testutils.GetFakeExecWithScripts(calls)
+	defer testutils.VerifyCalls(t, fexec, calls)
+	iptMgr := NewIptablesManager(fexec, NewFakeIptOperationShim())
 
-	if err = iptMgr.InitNpmChains(); err != nil {
-		t.Errorf("TestGetChainLineNumber @ iptMgr.InitNpmChains error: %s", err.Error())
-	}
-
-	entry = &IptEntry{
-		Chain: util.IptablesForwardChain,
-		Specs: []string{
-			util.IptablesJumpFlag,
-			util.IptablesAzureChain,
-		},
-	}
-
-	if npmExists, err = iptMgr.Exists(entry); err != nil {
-		t.Errorf("TestGetChainLineNumber failed @ azure-npm chain iptMgr.Exists error: %s", err.Error())
-	}
-
-	lineNum, err = iptMgr.GetChainLineNumber(util.IptablesAzureChain, util.IptablesForwardChain)
-	if err != nil {
-		t.Errorf("TestGetChainLineNumber @ after Init chains iptMgr.GetChainLineNumber error: %s", err.Error())
-	}
-
-	switch {
-	case (npmExists && kubeExists):
-		if lineNum < 2 {
-			t.Errorf("TestGetChainLineNumber @ after Init chains line number check iptMgr.GetChainLineNumber with npmExists: %t kubeExists: %t", npmExists, kubeExists)
-		}
-	case npmExists:
-		if lineNum == 0 {
-			t.Errorf("TestGetChainLineNumber @ after Init chains line number check iptMgr.GetChainLineNumber with npmExists: %t kubeExists: %t", npmExists, kubeExists)
-		}
-	case !npmExists:
-		t.Errorf("TestGetChainLineNumber @ after Init chains line number check iptMgr.GetChainLineNumber with failed to Add chain ")
-	}
-
-	if err = iptMgr.UninitNpmChains(); err != nil {
-		t.Errorf("TestGetChainLineNumber @ iptMgr.UninitNpmChains")
-	}
+	lineNum, err := iptMgr.GetChainLineNumber(util.IptablesAzureChain, util.IptablesForwardChain)
+	require.NoError(t, err)
+	require.Equal(t, lineNum, 3)
 }
 
 func TestMain(m *testing.M) {
 	metrics.InitializeAll()
-	iptMgr := NewIptablesManager()
-	iptMgr.Save(util.IptablesConfigFile)
+	metrics.CreateFakeTelemetryHandle()
```

Comment:
I think this is to avoid sending logs from functions (e.g., SendErrorLogAndMetric()) in UTs. Is it right? 

It is not scope of this PR and I am not familiar with metrics package code yet, but I feel like it would be nice to remove this dependency (e.g., metrics.InitializeAll()) to run UTs and revisit metrics package code within next few PRs to better structures if needed. Any thoughts?